File: src/main/java/dev/slimevr/vr/DeviceManager.java
package dev.slimevr.vr;

import dev.slimevr.VRServer;
import io.eiren.util.collections.FastList;


public class DeviceManager {

	private final FastList<IDevice> devices = new FastList<>();

	public FastList<IDevice> getDevices() {
		return devices;
	}

	private final VRServer server;

	public DeviceManager(VRServer server) {
		this.server = server;
	}

	public Device createDevice(String name, String version, String manufacturer) {
		Device device = new Device();

		device.setCustomName(name);
		device.setFirmwareVersion(version);
		device.setManufacturer(manufacturer);

		return device;
	}

	public void addDevice(IDevice device) {
		this.server.queueTask(() -> {
			this.devices.add(device);
		});
	}
}

File: server/core/src/main/java/dev/slimevr/tracking/trackers/DeviceManager.kt
package dev.slimevr.tracking.trackers

import dev.slimevr.VRServer
import io.eiren.util.collections.FastList

class DeviceManager(private val server: VRServer) {
	val devices = FastList<Device>()
	fun createDevice(origin: DeviceOrigin, name: String?, version: String?, manufacturer: String?): Device {
		val device = Device(origin)
		device.name = name
		device.firmwareVersion = version
		device.manufacturer = manufacturer
		return device
	}

	fun addDevice(device: Device) {
		server.queueTask { devices.add(device) }
	}
}
